src/OSPSuite.Core/Domain/Services/ObservedDataTask.cs

```diff
@@ -166,11 +178,73 @@ public void AddImporterConfigurationToProject(ImporterConfiguration configuratio
          _executionContext.AddToHistory(new AddImporterConfigurationToProjectCommand(configuration).Run(_executionContext));
       }
 
+      public void RemoveUsedObservedDataFromSimulation(IReadOnlyList<UsedObservedData> usedObservedDataList)
+      {
+         if (!usedObservedDataList.Any())
+            return;
+
+         foreach (var usedObservedData in usedObservedDataList)
+         {
+            var parameterIdentifications = findParameterIdentificationsUsing(usedObservedData).ToList();
+            if (!parameterIdentifications.Any())
+               continue;
+
+            _dialogCreator.MessageBoxInfo(
+               Captions.ParameterIdentification.CannotRemoveObservedDataBeingUsedByParameterIdentification(observedDataFrom(usedObservedData).Name,
+                  parameterIdentifications.AllNames().ToList()));
+            return;
+         }
+
+         var viewResult = _dialogCreator.MessageBoxYesNo(Captions.ReallyRemoveObservedDataFromSimulation);
+         if (viewResult == ViewResult.No)
+            return;
+
+         usedObservedDataList.GroupBy(x => x.Simulation).Each(x => removeUsedObservedDataFromSimulation(x, x.Key.DowncastTo<ISimulation>()));
```

Comment:
x.Key.DowncastTo<ISimulation>()

Why is this so? Are we sure of this at this stage?